Update sum.mdx docs (#33514)
hashicorp/terraform: website/docs/language/functions/sum.mdx

`sum` takes a list or set of numbers and returns the sum of those numbers.

> `sum` will fail if given an empty list or set.

Reviewer: ```suggestion
`sum` fails when given an empty list or set.
```
Use future tense per our style guide.
